db/active_replicator_common_collections.go

```diff
@@ -34,12 +34,23 @@ func (arc *activeReplicatorCommon) _initCollections() ([]replicationCheckpoint,
 		getCollectionsCheckpointIDs []string
 	)
 
+	// ensure remote collection set is the same length as local collection set
 	if remoteLen := len(arc.config.CollectionsRemote); remoteLen > 0 {
 		if localLen := len(arc.config.CollectionsLocal); localLen != remoteLen {
 			return nil, fmt.Errorf("local and remote collections must be the same length... had %d and %d", localLen, remoteLen)
 		}
 	}
 
+	// ensure channel filter set is the same length as local collection set
+	if collectionsChannelFilterLen := len(arc.config.CollectionsChannelFilter); collectionsChannelFilterLen > 0 {
+		if localLen := len(arc.config.CollectionsLocal); localLen != collectionsChannelFilterLen {
+			return nil, fmt.Errorf("local collections and channel filter set must be the same length... had %d and %d", localLen, collectionsChannelFilterLen)
+		}
+		if channelFilterLen := len(arc.config.FilterChannels); channelFilterLen > 0 {
+			return nil, fmt.Errorf("channel filter and collection channel filter set cannot both be set")
+		}
+	}
+
```

Comment:
nice to have because this function is getting massive, throw into a separate validation function.